src/store/common/reducer.ts

```diff
@@ -3,7 +3,10 @@ import { CommonState, Mode } from './types';
 import { toggleAnnotationModeAction, toggleAnnotationVisibilityAction } from './actions';
 
 const visibilityReducer = createReducer<CommonState['visibility']>(true, builder =>
-    builder.addCase(toggleAnnotationVisibilityAction, state => !state),
+    builder.addCase(
+        toggleAnnotationVisibilityAction,
+        (state, { payload: visibility }: { payload: boolean }) => visibility,
```

Comment:
We shouldn't need the `: boolean` type here, as it'll be inferred from the action.